app/actions/handlers.py

```diff
@@ -50,16 +51,31 @@ async def action_pull_observations(integration, action_config: PullObservationsC
 
     observations_extracted = 0
 
+    last_updated_time = await state_manager.get_state(
+        integration_id=integration.id,
+        action_id="pull_observations"
+    )
+
+    if not last_updated_time:
+        now = datetime.now(timezone.utc)
+        logger.info(f"Setting initial lookback hours to {action_config.look_back_window_hours} hrs from now")
+        start = (now - timedelta(hours=action_config.look_back_window_hours)).strftime("%Y-%m-%d %H:%M:%S")
+    else:
+        start = last_updated_time.get("last_updated_time")
+
     try:
-        logger.info(f"-- Getting observations for Username: {auth_config.username} --")
-        dataset = await client.get_observations(
+        logger.info(f"-- Getting observations for Username: {auth_config.username} from {start} --")
+        get_observations_response = await client.get_observations(
             url,
             username=auth_config.username,
             password=auth_config.password.get_secret_value(),
-            look_back_window_hours=int(action_config.look_back_window_hours)
+            start=start
         )
         
-        if dataset:
+        if get_observations_response:
```

Comment:
[nitpick] this is a nice place to use the `:=` operator to reduce the number of time that variable is typed.
```suggestion
        if get_observations_response := await client.get_observations(
            url,
            username=auth_config.username,
            password=auth_config.password.get_secret_value(),
            start=start
        )
```